ci/test/substitutions

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+set -ex
+# shellcheck source=./ci/test/common
+source ./ci/test/common
+init_test_repo
+cd "$test_repo_path"
+
+cat <<EOF > main.workflow
+workflow "wf" {
```

Comment:
we can remove a1, a2, c, and d actions from this and the two other workflows, as those are not relevant to this test. In addition, we can also check that substitutions are correctly changed in other parts of the workflow file ( in `env`, in `secrets`, etc..)

Reply:
Ok, I can fix it by adding the feature to substitute for the whole workflow.